### DISCUSSION TITLE
trouble with installation - Nominatim 3.7

### DISCUSSION THREAD

**Discussion by matro91** [osm-search/Nominatim#2266] [General]
I installed nomitaim following the instruction https://nominatim.org/release-docs/3.7.0/appendix/Install-on-Ubuntu-20/. Then i import the database as explain at https://nominatim.org/release-docs/3.7.0/admin/Import/ .
I 've done it twice, once i import the full planet, twice juste europe.
Il import the admin-style as i juste need city, not road etc...
the import is find, when it check, no error message, everything looks fine.
So i try to start the server 
nominatim serve and then lynx http://localhost:8088/status.php which return ok
but when I try lynx http://localhost:8088/search.php?q=Berlin i recieved an error 500 internal server error. If i try with debug=1

Debug output for Geocode::lookup

Geocode:  'Query' => 'Berlin'
          'Structured query' => False
          'Name keys' => 'name:en', 'name', 'brand', 'official_name:en', 'short_name:en', 'official_name', 'short_name', 'ref', 'type'
          'Excluded place IDs' =>
          'Try reversed query' => True
          'Limit (for searches)' => 20
          'Limit (for results)' => 10
          'Country codes' =>
          'Bounded search' => False
          'Viewbox' =>
          'Route points' =>
          'Route width' => False
          'Max rank' => 20
          'Min address rank' => 0
          'Max address rank' => 30
          'Address rank list' =>
     _________________________________________________________________________________________________________________________________

  Query Preprocessing

Normalized query:  'berlin'
Search context:  'Near radius' => False
                 'Near point (SQL)' => ''
                 'Bounded viewbox' => False
                 'Viewbox (SQL, small)' => ''
                 'Viewbox (SQL, large)' => ''
                 'Viewbox (SQL, centre)' => ''
                 'Countries (SQL)' => ''
                 'Excluded IDs (SQL)' => ''
Base search:  'Search rank' => 0
              'Country code' => ''
              'Name terms' =>
              'Name terms (stop words)' =>
              'Address terms' =>
              'Address terms (stop words)' =>
              'Address terms (full words)' => ''
              'Special search' => 0
              'Class' => ''
              'Type' => ''
              'House number' => ''
              'Postcode' => ''
Final query phrases:  0 => 'Berlin'
     _________________________________________________________________________________________________________________________________

  Tokenization

Tokens:  ' berlin' => ' berlin'
         'berlin' => 'berlin'

   SELECT word_id, word_token, word, class, type, country_code, operator, coalesce(search_name_count, 0) as count FROM word WHERE
   word_token in (' berlin','berlin')
   Valid Tokens:

         Group       ID Type Info
    berlin

   429021

   'word'

'partial' => False
'count' => 0

   berlin

   158052

   'word'

'partial' => True
'count' => 7514

   Phrases:

          Type       Phrase Words WordSets
   ''

   'berlin'

   0 => 'berlin'

   0 => 0 => 'berlin'
     _________________________________________________________________________________________________________________________________

  Search candidates

   rank    Name Tokens    Name Not Address Tokens Address Not country operator class type postcode housenumber
   1    # berlin(429021)#
   2    #berlin(158052)#
     _________________________________________________________________________________________________________________________________

  Search Loop, group 1, loop 1

   rank    Name Tokens    Name Not Address Tokens Address Not country operator class type postcode housenumber
   1    # berlin(429021)#

   SELECT place_id, address_rank, ( SELECT count(*) FROM ( SELECT unnest(ARRAY[429021]) INTERSECT SELECT unnest(nameaddress_vector) ) s)
   as exactmatch FROM search_name WHERE name_vector @> ARRAY[429021] ORDER BY (CASE WHEN importance = 0 OR importance IS NULL THEN
   0.75001-(search_rank::float/40) ELSE importance END) DESC, exactmatch DESC LIMIT 20
   {"error":{"code":500,"message":"Could not get places for search terms.","details":"/usr/local/lib/nominatim/lib-php/DB.php(187)"}}

. Thanks for your help
Mathilde

**Comment by lonvia** [comment]
Could you please have a look at the PostgreSQL logs (usually `/var/log/postgresql/postgresql-13-main.log`), if there is an error.

You could also see what happens if you run the bad query on the database directly: `psql -d nominatim -c 'SELECT place_id, address_rank, ( SELECT count(*) FROM ( SELECT unnest(ARRAY[429021]) INTERSECT SELECT unnest(nameaddress_vector) ) s)
as exactmatch FROM search_name WHERE name_vector @> ARRAY[429021] ORDER BY (CASE WHEN importance = 0 OR importance IS NULL THEN
0.75001-(search_rank::float/40) ELSE importance END) DESC, exactmatch DESC LIMIT 20'`

**Reply by matro91** [reply]
Hi
In /var/log/postgresql/postgresql-12-main.log i have this error :
 nominatim@nominatim ERROR:  relation "search_name" does not exist at character 162
2021-04-13 14:00:38.249 UTC [1381] nominatim@nominatim STATEMENT:  SELECT place_id, address_rank, (  SELECT count(*) FROM (   SELECT unnest(ARRAY[429021])    INTERSECT   SELECT unnest(nameaddress_vector) ) s) as exactmatch FROM search_name WHERE name_vector @> ARRAY[429021] ORDER BY (CASE WHEN importance = 0 OR importance IS NULL THEN 0.75001-(search_rank::float/40) ELSE importance END) DESC, exactmatch DESC LIMIT 20


i 've got the same error which the query on the database directly
Mathilde

**Reply by lonvia** [reply]
Have you imported with `--reverse-only`?

**Comment by matro91** [comment]
yes

**Reply by lonvia** [reply]
Well, there is your problem then. You can't use the forward query `search` when you have a reverse-only database. Try with `http://localhost:8088/reverse.php?lat=52&lon=1` and all will be well.

We probably should disable the `search` endpoint completely on a reverse-only database instead of sending a strange error message. I'll open a ticket about that.

**Reply by lonvia** [reply]
Done in issue #2267. (Including a hint to change the URL for testing the installation, which I presume is what you were trying to do.)
